### PR TITLE
feat: adding support for git CLI operation over go-git

### PR DESCRIPTION
**Pull request by subhashish-devtron**
1.  refactored flows to contain all go-git imports to one file GoGitUtil.go
2. refactored analytics code flows to RepositoryManagerAnalytics.go
3.  implemented CLI operations for flows except analytics code flows

### CODE REVIEW

File: pkg/git/Bean.go [hunk 1]
```diff
@@ -41,44 +44,146 @@ type CiPipelineMaterialBean struct {
...
+	git.Repository
+	rootDir     string
+	commitCount int
+}
+
+type CommitIterator interface {
+	Next() (GitCommit, error)
+}
+
+type CommitIteratorCli struct {
```
Comment (subhashish-devtron) [inline pkg/git/Bean.go:61]: Iterator at end

File: pkg/git/Bean.go [hunk 2]
```diff
@@ -41,44 +44,146 @@ type CiPipelineMaterialBean struct {
...
+	Date         time.Time
+	Message      string
+	Changes      []string     `json:",omitempty"`
+	FileStats    *FileStats   `json:",omitempty"`
+	WebhookData  *WebhookData `json:"webhookData"`
+	Excluded     bool         `json:",omitempty"`
+	CheckoutPath string
+}
+
+func (gitCommit *GitCommitBase) GetCommit() *GitCommitBase {
```
Comment (subhashish-devtron) [inline pkg/git/Bean.go:155]: is this needed?

File: pkg/git/Bean.go [hunk 3]
```diff
@@ -41,44 +44,146 @@ type CiPipelineMaterialBean struct {
...
+	commit, err := itr.CommitIter.Next()
+	gitCommit := GitCommitBase{
+		Author:  commit.Author.String(),
+		Commit:  commit.Hash.String(),
+		Date:    commit.Author.When,
+		Message: commit.Message,
+	}
+	return &GitCommitGoGit{
+		GitCommitBase: gitCommit,
+		cm:            commit,
```
Comment (subhashish-devtron) [inline pkg/git/Bean.go:106]: Kripansh to check

**Review by subhashish-devtron — COMMENTED** [review]
comments

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**